ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java

```diff
@@ -224,63 +232,122 @@ public void addHostToCluster(@EffectorParam(name = "Cluster name") String cluste
+            restAdapter.create(ServiceEndpoint.class).createComponent(cluster, service, mapping.getKey());
+            restAdapter.create(HostEndpoint.class).createHostComponent(cluster, mapping.getValue(), mapping.getKey());
+        }
+
+        // Step 4 - Create configuration, if needed
+        if (!StringUtils.isEmpty(configurationKey) && configuration != null) {
+            createServiceConfiguration(cluster, configurationKey, configuration);
+        }
+
+        // Step 5 - Install the service
+        Request request = restAdapter.create(ServiceEndpoint.class).updateService(cluster, service, ImmutableMap.builder()
+                .put("RequestInfo", ImmutableMap.builder()
+                        .put("context", String.format("Install %s service", service))
+                        .build())
+                .put("ServiceInfo", ImmutableMap.builder()
+                        .put("state", "INSTALLED")
+                        .build())
+                .build());
+        // Need to wait until the service is installed to start it. We leverage the HTTP feed to check the state of
+        // service installation. Once it's installed, we fire up the service start
+        final AttributeSensor<String> requestSensor = Sensors.newStringSensor(String.format("SERVICE_%s_INSTALL_STATE", service.toUpperCase()));
+        final SettableFuture<Boolean> requestFuture = SettableFuture.create();
+        final HttpFeed requestFeed = HttpFeed.builder()
+                .entity(this)
+                .period(1000, TimeUnit.MILLISECONDS)
+                .baseUri(request.getHref())
+                .credentials("admin", "admin")
+                .header(HttpHeaders.AUTHORIZATION, HttpTool.toBasicAuthorizationValue(usernamePasswordCredentials))
+                .poll(new HttpPollConfig<String>(requestSensor)
+                        .onSuccess(Functionals.chain(HttpValueFunctions.jsonContents(), getRequestState()))
+                        .onFailureOrException(Functions.<String>constant("FAILED")))
+                .build();
+        subscribe(this, requestSensor, new LockedEventListener(requestFuture) {
+            @Override
+            public void onEvent(SensorEvent<String> sensorEvent) {
+                if (StringUtils.isNotBlank(sensorEvent.getValue()) && !sensorEvent.getValue().equals("IN_PROGRESS")) {
+                    if (!sensorEvent.getValue().equals("COMPLETED")) {
+                        requestFuture.setException(new RuntimeException(String.format("Error during installation of service \"%s\". Please check the Ambari console for more details: %s", service, ambariUri)));
+                    } else {
+                        // Step 6 - Start the service
+                        startService(cluster, service);
+                        requestFuture.set(true);
+                    }
+                }
+            }
+        });
+
+        try {
+            requestFuture.get(5, TimeUnit.MINUTES);
```

Comment:
Yes, the thread is blocked because we need to wait for the operation to finish to report back. Not doing this means that as soon as we subscribe to the sensor, the effector finished, reporting `COMPLETED` within the Brooklyn UI which we obviously don't want